test/distributed/_tensor/test_dtensor.py

```diff
@@ -704,5 +707,265 @@ def test_split_tensor(self) -> None:
+
+    @property
+    def device_type(self) -> str:
+        return "cuda" if torch.cuda.is_available() else "cpu"
+
+    @property
+    def world_size(self) -> int:
+        return 2
+
+    def test_fakify_dtensor(self):
+        mesh = DeviceMesh(self.device_type, torch.arange(self.world_size))
+
+        # pass in DTensor as inputs/outputs to the function
+        def fn(x):
+            return x
+
+        x = DTensor.from_local(torch.rand(1), mesh, [Shard(0)], run_check=False)
+        ref = fn(x)
+
+        opt_fn = torch.compile(fn, backend="aot_eager", fullgraph=True)
+        res = opt_fn(x)
+        self.assertEqual(res, ref)
+
+    def test_dynamo_dtensor(self):
+        mesh = DeviceMesh(self.device_type, torch.arange(self.world_size))
+
+        # test passing in DTensor as inputs/outputs and run some tensor computation
+        def fn(x):
+            return x * x + 2
+
+        x = DTensor.from_local(torch.rand(1), mesh, [Shard(0)], run_check=False)
+        ref = fn(x)
+
+        opt_fn = torch.compile(fn, backend="aot_eager", fullgraph=True)
+        res = opt_fn(x)
+        self.assertEqual(res, ref)
+
+    def test_dynamo_dtensor_from_local(self):
+        mesh = DeviceMesh(self.device_type, torch.arange(self.world_size))
+
+        # create DTensor inside fn and run some compute
+        def fn(x):
+            dt = DTensor.from_local(x, mesh, [Replicate()], run_check=False)
+            return dt.to_local() + 2
+
+        # below is the op approach for reference
+        # from torch.distributed._tensor.api import _FromTorchTensor
+        # def from_local_tensor(x):
+        #     return _FromTorchTensor.apply(x, mesh, [Replicate()], False)
+
+        # _dt_lib_def = torch.library.Library("dtensor", "DEF")
+        # _dt_lib_def.define("from_local(Tensor self) -> Tensor")
+
+        # _dt_lib_impl = torch.library.Library("dtensor", "IMPL")
+        # _dt_lib_impl.impl("from_local", from_local_tensor, "Autograd")
+
+        x = torch.ones(1)
+        ref = fn(x)
+        opt_fn = torch.compile(fn, backend="aot_eager", fullgraph=True)
+        res = opt_fn(x)
+        self.assertEqual(res, ref)
+
+    def test_dynamo_dtensor_from_local_redistribute(self):
+        mesh = DeviceMesh(self.device_type, torch.arange(self.world_size))
+
+        # pass in tensor as inputs/outputs, create DTensor and run redistribute
+        # (allgather collective) inside the fn
+        def fn(x):
+            dt = DTensor.from_local(x, mesh, [Shard(0)], run_check=False)
+            return dt.redistribute(mesh, [Replicate()]).to_local() + 2
+
+        x = torch.ones(1)
+        ref = fn(x)
+        opt_fn = torch.compile(fn, backend="aot_eager", fullgraph=True)
+        res = opt_fn(x)
+        self.assertEqual(res, ref)
+
+    # This is a test where we do the conversions to-from DTensors *outside* of the compiled region,
+    # and the compiled graph has DTensor inputs and outputs.
+    def test_compile_dtensor_simple(self):
```

Comment:
can we move these test cases to `test_dtensor_compile.py` so that we can group all pt2 related tests? thanks!

Reply:
will do!